### ISSUE TITLE
Pubsub disco#info for XEP-0084: User Avatar not meeting XEP-60 5.3 requirements

### ISSUE THREAD

**Issue by cmeng-git** [processone/ejabberd#1895]
> What version of ejabberd are you using?

17.06

> What operating system (version) are you using?

Ubuntu 16.04

> How did you install ejabberd (source, package, distribution)?

source built

> What did not work as expected? Are there error messages in the log? What

aTalk, an xmpp client for android using library smack 4.2.1-beta2-SNAPSHOT. While performing an pubsub update for AvatarData and AvatarMetadata using the smack PubSubManager, PubSubManager throws an exception indicating reply stanza for disco#info not meeting  XEP-60 5.3 as attached below:

> XEP-0060: Publish-Subscribe - XMPP
> 5.3 Discover Node Information
> 
> A pubsub service MUST allow entities to query individual nodes for the information associated with that node. The Service Discovery protocol MUST be used to discover this information. The "disco#info" result MUST include an identity with a category of "pubsub" and a type of either "leaf" or "collection".
> 
> Note: If a node has an identity type of "leaf", then it MUST NOT contain other nodes or collections (only items); if a node has an identity type of "collection", then it MUST NOT contain items (only other nodes or collections).

ejabberd 17.06 xmpp server ejabberd.yml configuration
```yaml
  mod_pubsub: 
    db_type: sql
    access_createnode: pubsub_createnode
    ignore_pep_from_offline: true
    last_item_cache: false
    plugins:
      - "flat"
      - "hometree"
      - "pep"
```
aTalk log for AvatarData reply
```
07-31 08:46:26.984 D/SMACK: SENT (0): <iq to='leopard@atalk.org' id='5rW2U-134' type='get'><query xmlns='http://jabber.org/protocol/disco#info' node='urn:xmpp:avatar:data'></query></iq>
07-31 08:46:27.004 D/SMACK: RECV (0): <iq xml:lang='en' to='leopard@atalk.org/atalk' from='leopard@atalk.org' type='result' id='5rW2U-134'><query node='urn:xmpp:avatar:data' xmlns='http://jabber.org/protocol/disco#info'><identity type='registered' category='account'/></query></iq>
07-31 08:46:27.044 E/αTalk: [1] util.UtilActivator.uncaughtException().95 An uncaught exception occurred in thread=Thread[main,5,main] and message was: PubSub service 'leopard@atalk.org' returned disco info result for node 'urn:xmpp:avatar:data', but it did not contain an Identity of type 'leaf' or 'collection' (and category 'pubsub'), which is not allowed according to XEP-60 5.3.
                            org.jivesoftware.smackx.pubsub.PubSubAssertionError$DiscoInfoNodeAssertionError: PubSub service 'leopard@atalk.org' returned disco info result for node 'urn:xmpp:avatar:data', but it did not contain an Identity of type 'leaf' or 'collection' (and category 'pubsub'), which is not allowed according to XEP-60 5.3.
                                at org.jivesoftware.smackx.pubsub.PubSubManager.getNode(PubSubManager.java:255)
                                at org.jivesoftware.smackx.pep.PEPManager.publish(PEPManager.java:145)
                                at org.jivesoftware.smackx.avatar.useravatar.UserAvatarManager.publishAvatarData(UserAvatarManager.java:188)
                                at org.jivesoftware.smackx.avatar.useravatar.UserAvatarManager.publishAvatarData(UserAvatarManager.java:168)
```

> was the unexpected behavior? What was the expected result?

Not implemented as per XEP-60 section 5.3. requirements OR need changes to ejabberd.yml configuration settings?

**Comment by cromain** [comment]
This should be fixed by #1717 
Note: as of chapter 6.1 of XEP-0163, the disco info result includes type 'pep' for category 'pubsub'

**Comment by jigsp269** [comment]
Please provide the solution for this 
I am configure ejabberd 18.04 and configure **mod_pubsub** as above 
**mod_pubsub: 
    db_type: sql
    access_createnode: pubsub_createnode
    ignore_pep_from_offline: true
    last_item_cache: false
    plugins:
      - "flat"
      - "hometree"
      - "pep"**

and also add 
**mod_vcard_xupdate:
 use_cache: false**

I want to add or update profile picture to client 
I use client as a conversation app.
On app side i get error **"your server does not support the publication of avatars"**
